packages/firebase_auth/lib/firebase_auth.dart

```diff
@@ -192,4 +223,24 @@ class FirebaseAuth {
     final FirebaseUser currentUser = new FirebaseUser._(data);
     return currentUser;
   }
+
+  Future<Null> _callHandler(MethodCall call) async {
+    switch (call.method) {
+      case "onAuthStateChanged":
+        _onAuthStageChangedHandler(call);
+        break;
+    }
+    return null;
+  }
+
+  void _onAuthStageChangedHandler(MethodCall call) {
+    final Map<String, dynamic> data = call.arguments;
+    final FirebaseUser currentUser =
+        data != null ? new FirebaseUser._(data) : null;
+    _authStateChangedController
```

Comment:
Rather than iterating through all the controllers, just look up the one controller identified by the `identifier` index and fire an event on that one. This ensures that listening to one stream doesn't trigger events being fired on another stream.